### ISSUE TITLE
[MAIN] [Pipeline] Model tracking leaderboard

### ISSUE THREAD

**Issue by Mars-Martiny** [Mars-Martiny/CITS5206_group1#27]
Writes to singular csv which tracks the performance/evaluation metrics of the models and location at which each model is saved.

Potentially also track:
- hyperparameters used to train it -> useful for side-by-side comparisons in fine-tuning stage 
- pipeline specifics (non-hyperparameter selections made such as model architecture, loss function selection, etc.) 
- the pipeline version (referencing exact point in repository timeline) -> useful if large changes to overall pipeline are made

**Comment by AndreHrs** [comment]
I will assign this to me since I have a model running already and it will be easy for me to get it going!
